Mathlib/AlgebraicGeometry/EllipticCurve/DivisionPolynomial/Basic.lean

```diff
@@ -0,0 +1,427 @@
+
+/-- The univariate polynomial $\tilde{\Psi}_4$, which is auxiliary to the $4$-division polynomial
+$\psi_4 = \Psi_4 = \tilde{\Psi}_4\psi_2$. -/
+protected noncomputable def preΨ₄ : R[X] :=
+  2 * X ^ 6 + C W.b₂ * X ^ 5 + 5 * C W.b₄ * X ^ 4 + 10 * C W.b₆ * X ^ 3 + 10 * C W.b₈ * X ^ 2 +
+    C (W.b₂ * W.b₈ - W.b₄ * W.b₆) * X + C (W.b₄ * W.b₈ - W.b₆ ^ 2)
+
+/-- The univariate polynomials $\tilde{\Psi}_n$ for $n \in \mathbb{N}$, which are auxiliary to the
+bivariate polynomials $\Psi_n$ congruent to the bivariate $n$-division polynomials $\psi_n$. -/
+protected noncomputable def preΨ' (n : ℕ) : R[X] :=
+  preNormEDS' (W.Ψ₂Sq ^ 2) W.Ψ₃ W.preΨ₄ n
+
+@[simp]
+lemma preΨ'_zero : W.preΨ' 0 = 0 :=
+  preNormEDS'_zero ..
+
+@[simp]
+lemma preΨ'_one : W.preΨ' 1 = 1 :=
+  preNormEDS'_one ..
+
+@[simp]
+lemma preΨ'_two : W.preΨ' 2 = 1 :=
+  preNormEDS'_two ..
+
+@[simp]
+lemma preΨ'_three : W.preΨ' 3 = W.Ψ₃ :=
+  preNormEDS'_three ..
+
+@[simp]
+lemma preΨ'_four : W.preΨ' 4 = W.preΨ₄ :=
+  preNormEDS'_four ..
+
+lemma preΨ'_odd (m : ℕ) : W.preΨ' (2 * (m + 2) + 1) =
+    W.preΨ' (m + 4) * W.preΨ' (m + 2) ^ 3 * (if Even m then W.Ψ₂Sq ^ 2 else 1) -
+      W.preΨ' (m + 1) * W.preΨ' (m + 3) ^ 3 * (if Even m then 1 else W.Ψ₂Sq ^ 2) :=
+  preNormEDS'_odd ..
+
+lemma preΨ'_even (m : ℕ) : W.preΨ' (2 * (m + 3)) =
+    W.preΨ' (m + 2) ^ 2 * W.preΨ' (m + 3) * W.preΨ' (m + 5) -
+      W.preΨ' (m + 1) * W.preΨ' (m + 3) * W.preΨ' (m + 4) ^ 2 :=
+  preNormEDS'_even ..
+
+/-! ### The univariate polynomials $\tilde{\Psi}_n$ for $n \in \mathbb{Z}$ -/
+
+/-- The univariate polynomials $\tilde{\Psi}_n$ for $n \in \mathbb{Z}$, which are auxiliary to the
+bivariate polynomials $\Psi_n$ congruent to the bivariate $n$-division polynomials $\psi_n$. -/
+protected noncomputable def preΨ (n : ℤ) : R[X] :=
+  preNormEDS (W.Ψ₂Sq ^ 2) W.Ψ₃ W.preΨ₄ n
+
+@[simp]
+lemma preΨ_ofNat (n : ℕ) : W.preΨ n = W.preΨ' n :=
+  preNormEDS_ofNat ..
+
+@[simp]
+lemma preΨ_zero : W.preΨ 0 = 0 :=
+  preNormEDS_zero ..
+
+@[simp]
+lemma preΨ_one : W.preΨ 1 = 1 :=
+  preNormEDS_one ..
+
+@[simp]
+lemma preΨ_two : W.preΨ 2 = 1 :=
+  preNormEDS_two ..
+
+@[simp]
+lemma preΨ_three : W.preΨ 3 = W.Ψ₃ :=
+  preNormEDS_three ..
+
+@[simp]
+lemma preΨ_four : W.preΨ 4 = W.preΨ₄ :=
+  preNormEDS_four ..
+
+lemma preΨ_odd (m : ℕ) : W.preΨ (2 * (m + 2) + 1) =
+    W.preΨ (m + 4) * W.preΨ (m + 2) ^ 3 * (if Even m then W.Ψ₂Sq ^ 2 else 1) -
+      W.preΨ (m + 1) * W.preΨ (m + 3) ^ 3 * (if Even m then 1 else W.Ψ₂Sq ^ 2) :=
+  preNormEDS_odd ..
+
+lemma preΨ_even (m : ℕ) : W.preΨ (2 * (m + 3)) =
+    W.preΨ (m + 2) ^ 2 * W.preΨ (m + 3) * W.preΨ (m + 5) -
+      W.preΨ (m + 1) * W.preΨ (m + 3) * W.preΨ (m + 4) ^ 2 :=
+  preNormEDS_even ..
+
+@[simp]
+lemma preΨ_neg (n : ℤ) : W.preΨ (-n) = -W.preΨ n :=
+  preNormEDS_neg ..
+
+/-! ### The univariate polynomials $\Psi_n^{[2]}$ -/
+
+/-- The univariate polynomials $\Psi_n^{[2]}$ congruent to $\psi_n^2$. -/
+protected noncomputable def ΨSq (n : ℤ) : R[X] :=
```

Comment:
I think only names of a single letter need to be protected. Maybe also those with one subscript, like `Ψ₃` and `ψ₂`. I don't see why `ΨSq` needs to be protected since conflicts seem unlikely.